### ISSUE TITLE
mod_wsgi-4.2.7 doesn't compile with Python 2.4

### ISSUE THREAD

**Issue by sitz** [GrahamDumpleton/mod_wsgi#24]
I was trying to install mod_wsgi-4.2.7 from source using Python 2.4, And It fails at during `make` execution.
Also, It works with Python 2.7 just fine.

Are these versions incompatible by design or it's bug?
Can someone please look into it?

This is the log it generates:

```
./configure --with-python=/usr/bin/python2.4 --with-apxs=/usr/sbin/apxs

checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for prctl... yes
checking Apache version... 2.2.3
configure: creating ./config.status
config.status: creating Makefile
```

```
make

/usr/sbin/apxs -c -I/usr/include/python2.4 -DNDEBUG -D_GNU_SOURCE  -Wc,-g -Wc,-O2  src/server/mod_wsgi.c src/server/wsgi_*.c -L/usr/lib -L/usr/lib/python2.4/config  -lpython2.4 -lpthread -ldl  -lutil -lm
/usr/lib/apr-1/build/libtool --silent --mode=compile gcc -prefer-pic -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m32 -march=i386 -mtune=generic -fasynchronous-unwind-tables -fno-strict-aliasing  -DLINUX=2 -D_REENTRANT -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -pthread -I/usr/include/httpd  -I/usr/include/apr-1   -I/usr/include/apr-1  -g -O2 -I/usr/include/python2.4 -DNDEBUG -D_GNU_SOURCE  -c -o src/server/mod_wsgi.lo src/server/mod_wsgi.c && touch src/server/mod_wsgi.slo
src/server/mod_wsgi.c: In function 'wsgi_log_stack_traces':
src/server/mod_wsgi.c:8107: warning: implicit declaration of function '_PyThread_CurrentFrames'
src/server/mod_wsgi.c:8107: warning: assignment makes pointer from integer without a cast
src/server/mod_wsgi.c:8118: error: 'Py_ssize_t' undeclared (first use in this function)
src/server/mod_wsgi.c:8118: error: (Each undeclared identifier is reported only once
src/server/mod_wsgi.c:8118: error: for each function it appears in.)
src/server/mod_wsgi.c:8118: error: expected ';' before 'i'
src/server/mod_wsgi.c:8124: error: 'i' undeclared (first use in this function)
apxs:Error: Command failed with rc=65536
.
make: *** [src/server/mod_wsgi.la] Error 1
```

**Comment by soumen0** [comment]
I m running Debian 9 stretch, Python 3.5. I have Python 3.5-dev installed also. But make gives me following error...

friday@debianLabV1:~/mod_wsgi-4.6.5$ sudo make
[sudo] password for friday:
/usr/bin/apxs2 -c  -Wc,-g -Wc,-O2  src/server/mod_wsgi.c src/server/wsgi_*.c -L -L  -lpython
/usr/share/apr-1.0/build/libtool  --mode=compile --tag=disable-static x86_64-linux-gnu-gcc -prefer-pic -pipe -g -O2 -fdebug-prefix-map=/build/apache2-2.4.25=. -fstack-protector-strong -Wformat -Werror=format-security  -DBUILD_DATETIME='"2018-11-03T18:46:19"' -Wdate-time -D_FORTIFY_SOURCE=2   -DLINUX -D_REENTRANT -D_GNU_SOURCE  -pthread  -I/usr/include/apache2  -I/usr/include/apr-1.0   -I/usr/include/apr-1.0 -I/usr/include -g -O2  -c -o src/server/mod_wsgi.lo src/server/mod_wsgi.c && touch src/server/mod_wsgi.slo
libtool: compile:  x86_64-linux-gnu-gcc -pipe -g -O2 -fdebug-prefix-map=/build/apache2-2.4.25=. -fstack-protector-strong -Wformat -Werror=format-security -DBUILD_DATETIME=\"2018-11-03T18:46:19\" -Wdate-time -D_FORTIFY_SOURCE=2 -DLINUX -D_REENTRANT -D_GNU_SOURCE -pthread -I/usr/include/apache2 -I/usr/include/apr-1.0 -I/usr/include/apr-1.0 -I/usr/include -g -O2 -c src/server/mod_wsgi.c  -fPIC -DPIC -o src/server/.libs/mod_wsgi.o
In file included from src/server/mod_wsgi.c:22:0:
src/server/wsgi_python.h:24:20: fatal error: Python.h: No such file or directory
 #include <Python.h>
                    ^
compilation terminated.
apxs:Error: Command failed with rc=65536
.
Makefile:31: recipe for target 'src/server/mod_wsgi.la' failed
make: *** [src/server/mod_wsgi.la] Error 1

**Comment by GrahamDumpleton** [comment]
Please don't add new problems against existing issues, create a new issue.

At a guess the problem is that you haven't told ``configure`` to use ``python3`` and it is still using ``python`` which is Python 2. Supply ``--with-python python3`` to ``configure``. Make sure you run ``make distclean`` first to clear out results of prior build attempt.

If this doesn't solve the issue, create a new issue, please don't use this closed issue.

**Comment by soumen0** [comment]
Actually I did the compilation with 3.5 only and since there was no error, I didn't mentioned it.

Here is the output in detail....
================================================================
friday@debianLabV1:~/mod_wsgi-4.6.5$ sudo ./configure --with-python=/usr/lib/python3.5
[sudo] password for friday:
checking for apxs2... /usr/bin/apxs2
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for prctl... yes
checking Apache version... 2.4.25
./configure: line 2849: /usr/lib/python3.5: Is a directory
./configure: line 2853: /usr/lib/python3.5: Is a directory
./configure: line 2861: /usr/lib/python3.5: Is a directory
./configure: line 2866: /usr/lib/python3.5: Is a directory
./configure: line 2880: /usr/lib/python3.5: Is a directory
./configure: line 2884: /usr/lib/python3.5: Is a directory
./configure: line 2887: /usr/lib/python3.5: Is a directory
./configure: line 2890: /usr/lib/python3.5: Is a directory
./configure: line 2893: /usr/lib/python3.5: Is a directory
./configure: line 2919: /usr/lib/python3.5: Is a directory
./configure: line 2922: /usr/lib/python3.5: Is a directory
configure: creating ./config.status
config.status: creating Makefile
friday@debianLabV1:~/mod_wsgi-4.6.5$ make
/usr/bin/apxs2 -c  -Wc,-g -Wc,-O2  src/server/mod_wsgi.c src/server/wsgi_*.c -L -L  -lpython
/usr/share/apr-1.0/build/libtool  --mode=compile --tag=disable-static x86_64-linux-gnu-gcc -prefer-pic -pipe -g -O2 -fdebug-prefix-map=/build/apache2-2.4.25=. -fstack-protector-strong -Wformat -Werror=format-security  -DBUILD_DATETIME='"2018-11-03T18:46:19"' -Wdate-time -D_FORTIFY_SOURCE=2   -DLINUX -D_REENTRANT -D_GNU_SOURCE  -pthread  -I/usr/include/apache2  -I/usr/include/apr-1.0   -I/usr/include/apr-1.0 -I/usr/include -g -O2  -c -o src/server/mod_wsgi.lo src/server/mod_wsgi.c && touch src/server/mod_wsgi.slo
libtool: compile:  x86_64-linux-gnu-gcc -pipe -g -O2 -fdebug-prefix-map=/build/apache2-2.4.25=. -fstack-protector-strong -Wformat -Werror=format-security -DBUILD_DATETIME=\"2018-11-03T18:46:19\" -Wdate-time -D_FORTIFY_SOURCE=2 -DLINUX -D_REENTRANT -D_GNU_SOURCE -pthread -I/usr/include/apache2 -I/usr/include/apr-1.0 -I/usr/include/apr-1.0 -I/usr/include -g -O2 -c src/server/mod_wsgi.c  -fPIC -DPIC -o src/server/.libs/mod_wsgi.o
In file included from src/server/mod_wsgi.c:22:0:
src/server/wsgi_python.h:24:20: fatal error: Python.h: No such file or directory
 #include <Python.h>
                    ^
compilation terminated.
apxs:Error: Command failed with rc=65536
.
Makefile:31: recipe for target 'src/server/mod_wsgi.la' failed
make: *** [src/server/mod_wsgi.la] Error 1

**Comment by soumen0** [comment]
Reply actually I am configuring with Python 3.5. I have created a [new issue](https://github.com/GrahamDumpleton/mod_wsgi/issues/399) . Details are there
